x-pack/plugins/alerting/server/task_runner/task_runner.ts

```diff
@@ -750,10 +814,11 @@ export class TaskRunner<
     let stateWithMetrics: Result<RuleTaskStateAndMetrics, Error>;
     let schedule: Result<IntervalSchedule, Error>;
     try {
-      const preparedResult = await this.timer.runWithTimer(
-        TaskRunnerTimerSpan.PrepareRule,
-        async () => this.prepareToRun()
-      );
+      if (!this.ruleData) {
```

Comment:
`this.ruleData` only seems to be set in `this.beforeRun()`, so I'm not seeing a case where we need the `if`.

Reply:
Just in case of the run method is called directly by something other than TM task cycle, such as ephemeral task runner, or tests.

Reply:
ruleData is different than the return value of beforeRun.

It may load and format more data than the rawRule (or rawAction in actions) but the TaskManager expects just the rawRule.